overlord/state/taskrunner.go

```diff
@@ -258,9 +260,7 @@ func (r *TaskRunner) run(t *Task) {
 				// Would work without it but might take two ensures.
 				r.tryUndo(t)
 			} else {
-				// XXX: Should we allow supplying a state with the Wait error?
-				// Currently using DoneStatus as that was the previous behaviour.
-				t.Wait(DoneStatus)
+				t.SetToWait(x.WaitedStatus)
```

Comment:
we probably should pick DoneStatus if WaitedStatus is DefaultStatus, with a test

Reply:
Makes a lot of sense - added this!